src/__tests__/preset-by-number.test.ts

```diff
@@ -0,0 +1,115 @@
+import { describe, expect, test } from '@jest/globals'
+import { parsePresetVariableOption } from '../preset-by-number.js'
+import { CompanionOptionValues } from '@companion-module/base'
+import { PresetRecall } from '../camera/commands.js'
+import { prettyBytes } from '../visca/utils.js'
+import { MockContext } from './mock-context.js'
+import type { MatcherFunction } from 'expect'
+import { Command } from '../visca/command.js'
+
+function optionsWithPresetOption(text: string): CompanionOptionValues {
+	return { val: text }
+}
+
+/**
+ * Custom jest matcher to check that options are not a string and that the command bytes match
+ * @param actual
+ * @param command
+ * @param expected_bytes
+ * @returns
+ */
+const toBeExpectedCommandBytes: MatcherFunction<[command: Command, expected_bytes: string]> = function (
+	actual,
+	command,
+	expected_bytes
+) {
+	if (typeof actual === 'string') {
+		throw new TypeError('Input showuld be of type CompanionOptionValues not string')
+	}
+
+	const commandBytes = command.toBytes(actual as CompanionOptionValues)
+	const pass = prettyBytes(commandBytes) === expected_bytes
+
+	return {
+		message: () =>
+			`expected ${this.utils.printReceived(prettyBytes(commandBytes))} ${
+				pass ? 'not ' : ''
+			}to match\nreceived ${this.utils.printExpected(expected_bytes)}`,
+		pass: pass,
+	}
+}
+
+/**
+ * Custom jest matcher to ensure that options is a string and has appropriate error
+ * @param actual
+ * @returns
+ */
+const toBeErrorString: MatcherFunction<[]> = function (actual) {
+	let pass = false
+	if (typeof actual === 'string') {
+		pass = actual.includes('Invalid recall preset value of')
+	}
+
+	return {
+		message: () => `expected result ${pass ? 'not ' : ''}to be an error string, received: ${actual}`,
+		pass: pass,
+	}
+}
+
+declare module 'expect' {
```

Comment:
I gotta be honest, I've been sitting and staring at this _mostly_ because I'm pretty leery about spinning up an entire custom matcher just for one function return type.  The closer we stick to relatively vanilla JS/TypeScript -- for example, writing helper functions to perform tests of all type-arms -- and the further from delving into test framework intricacies, the more understandable it all stays for Jest n00bs (like, um, me :innocent:).

But maybe I've stared at this enough to come around on it.

That said...this `declare module` goo is so much opaque "is it correct?  :shrug:" for me, that makes me hesitate...

Reply:
From the jest examples for custom matcher  https://github.com/jestjs/jest/blob/bd1c6db7c15c23788ca3e09c919138e48dd3b28a/examples/expect-extend/toBeWithinRange.ts